### DISCUSSION TITLE
Lesson 11 Part 23 | cast sig with _$ characters return different results. Why?

### DISCUSSION THREAD

**Discussion by veebull** [Cyfrin/foundry-full-course-cu#948] [General]
When we went for [https://openchain.xyz/signatures](https://openchain.xyz/signatures) to decode `0x23b872dd`
I found that `__$_$__$$$$$__$$_$$$_$$__$$___$$(address,address,uint256)` is also equal `0x23b872dd`

then i tried to encode it with cast sig and i get always different results. Check yourself.

`cast sig "__$_$__$$$$$__$$_$$$_$$__$$___$$(address,address,uint256)"`
> 0xba3f71a9

`cast sig "__$_$__$$$$$__$$_$$$_$$__$$___$$(address,address,uint256)"`
> 0x0bd6065e

`cast sig "__$_$__$$$$$__$$_$$$_$$__$$___$$(address,address,uint256)"`
> 0x61b8db71

... etc

Why does this happen? I thought the `sig` command must return a static value.

**Comment by shikhar229169** [comment]
I wonder why it is producing unique value everytime. But on other hand I tried it to do with taking the first 4 bytes from its keccak256 and it is giving the same result everytime:
<br>
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/111875681/f8cd5ca3-afdc-48b7-97d6-a7eb9e1aeb0c)

**Reply by veebull** [reply]
sounds like bug in foundry, submitted an issue [there](https://github.com/foundry-rs/book/issues/1019)

**Comment by theirrationalone** [comment]
Maybe, some kind of interpolation bug you have encountered, what i think is, if we, just for experiment, remove those leading underscores and start the string with most significant char $. Maybe that could resolve ambiguity..?!?!? 😅

`_` & `$` both are special keywords btw.

**Comment by veebull** [comment]
So after openning issue on foundry, I've got the answer [here](https://github.com/foundry-rs/foundry/issues/5970#issuecomment-1742082714). 
If use single quotes instead double, it will show result correctly. It is about how variables work with special characters (_$) in shell. I'm still figuring it out.
